plugin/src/main/java/org/autorefactor/refactoring/rules/AndroidWakeLockRefactoring.java

```diff
@@ -0,0 +1,152 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program under LICENSE-GNUGPL.  If not, see
+ * <http://www.gnu.org/licenses/>.
+ *
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution under LICENSE-ECLIPSE, and is
+ * available at http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.autorefactor.refactoring.rules;
+
+import static org.autorefactor.refactoring.ASTHelper.DO_NOT_VISIT_SUBTREE;
+import static org.autorefactor.refactoring.ASTHelper.VISIT_SUBTREE;
+import static org.autorefactor.refactoring.ASTHelper.isMethod;
+
+import java.util.Collections;
+
+import static org.autorefactor.refactoring.ASTHelper.getAncestor;
+
+import org.autorefactor.refactoring.ASTBuilder;
+import org.autorefactor.refactoring.FinderVisitor;
+import org.autorefactor.refactoring.Refactorings;
+import org.eclipse.jdt.core.dom.ASTNode;
+import org.eclipse.jdt.core.dom.IMethodBinding;
+import org.eclipse.jdt.core.dom.MethodDeclaration;
+import org.eclipse.jdt.core.dom.MethodInvocation;
+import org.eclipse.jdt.core.dom.SingleVariableDeclaration;
+import org.eclipse.jdt.core.dom.Statement;
+import org.eclipse.jdt.core.dom.TypeDeclaration;
+
+/** See {@link #getDescription()} method. */
+public class AndroidWakeLockRefactoring extends AbstractRefactoringRule {
+
+    @Override
+    public String getDescription() {
+        return "Failing to release a wakelock properly can keep the Android device in a high "
+            + "power mode, which reduces battery life. There are several causes of this, such "
+            + "as releasing the wake lock in onDestroy() instead of in onPause(), failing to "
+            + "call release() in all possible code paths after an acquire(), and so on.";
+    }
+
+    @Override
+    public String getName() {
+        return "Android WakeLock";
+    }
+
+    @Override
+    public boolean visit(MethodInvocation node) {
+        if (isMethod(node, "android.os.PowerManager.WakeLock", "release")) {
+            // check whether it is being called in onDestroy
+            MethodDeclaration enclosingMethod = getAncestor(node, MethodDeclaration.class);
+            if (isMethod(enclosingMethod.resolveBinding(), "android.app.Activity", "onDestroy")) {
+                final Refactorings r = this.ctx.getRefactorings();
+                TypeDeclaration typeDeclaration = getAncestor(enclosingMethod, TypeDeclaration.class);
+                MethodDeclaration onPauseMethod = findMethod(typeDeclaration, "onPause");
+                if (onPauseMethod != null && node.getParent().getNodeType() == ASTNode.EXPRESSION_STATEMENT) {
+                    r.remove(node.getParent());
```

Comment:
What is the parent node? What do you expect to remove here?
That seems risky to removing something without checking what it is.